libraries/abstractions/transport/utest/transport_secure_sockets_utest.c

```diff
@@ -0,0 +1,740 @@
+    };
+
+    SOCKETS_Socket_ExpectAnyArgsAndReturn( SOCKETS_ERROR_NONE );
+    SOCKETS_SetSockOpt_ExpectAnyArgsAndReturn( SOCKETS_ERROR_NONE );
+    SOCKETS_GetHostByName_ExpectAnyArgsAndReturn( MOCK_SERVER_ADDRESS );
+    SOCKETS_Connect_ExpectAnyArgsAndReturn( SOCKETS_ERROR_NONE );
+    SOCKETS_SetSockOpt_ExpectAnyArgsAndReturn( SOCKETS_ERROR_NONE );
+    SOCKETS_SetSockOpt_ExpectAnyArgsAndReturn( SOCKETS_ERROR_NONE );
+    returnStatus = SecureSocketsTransport_Connect( &networkContext,
+                                                   &serverInfo,
+                                                   &localSocketsConfig );
+    TEST_ASSERT_EQUAL( TRANSPORT_SOCKET_STATUS_SUCCESS, returnStatus );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Test #SecureSocketsTransport_Disconnect with invalid parameters.
+ */
+void test_SecureSocketsTransport_Disconnect_NULL_Network_Context( void )
+{
+    TransportSocketStatus_t returnStatus;
+
+    returnStatus = SecureSocketsTransport_Disconnect( NULL );
+    TEST_ASSERT_EQUAL( TRANSPORT_SOCKET_STATUS_INVALID_PARAMETER, returnStatus );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Test the #SecureSocketsTransport_Disconnect and encounterd #SOCKETS_Shutdown failure.
+ */
+void test_SecureSocketsTransport_Disconnect_Fail_to_ShutDown( void )
+{
+    TransportSocketStatus_t returnStatus;
+
+    SOCKETS_Shutdown_ExpectAnyArgsAndReturn( MOC_SECURE_SOCKET_ERROR );
+    returnStatus = SecureSocketsTransport_Disconnect( &networkContext );
+    TEST_ASSERT_EQUAL( TRANSPORT_SOCKET_STATUS_INTERNAL_ERROR, returnStatus );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Test #SecureSocketsTransport_Disconnect and encounterd #SOCKETS_Close failure.
+ */
+void test_SecureSocketsTransport_Disconnect_Fail_to_Close( void )
+{
+    TransportSocketStatus_t returnStatus;
+
+    SOCKETS_Shutdown_ExpectAnyArgsAndReturn( SOCKETS_ERROR_NONE );
+    SOCKETS_Close_ExpectAnyArgsAndReturn( MOC_SECURE_SOCKET_ERROR );
+    returnStatus = SecureSocketsTransport_Disconnect( &networkContext );
+    TEST_ASSERT_EQUAL( TRANSPORT_SOCKET_STATUS_INTERNAL_ERROR, returnStatus );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Happy path #SecureSocketsTransport_Disconnect.
+ */
+void test_SecureSocketsTransport_Disconnect_Sucess( void )
+{
+    TransportSocketStatus_t returnStatus;
+
+    SOCKETS_Shutdown_ExpectAnyArgsAndReturn( SOCKETS_ERROR_NONE );
+    SOCKETS_Close_ExpectAnyArgsAndReturn( SOCKETS_ERROR_NONE );
+    returnStatus = SecureSocketsTransport_Disconnect( &networkContext );
+    TEST_ASSERT_EQUAL( TRANSPORT_SOCKET_STATUS_SUCCESS, returnStatus );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Test that #SecureSocketsTransport_Send is able to return that 0 bytes are sent over
+ * the network stack when passing any invalid parameters.
```

Comment:
If the sentence can survive without a certain clause, then it should have a comma around the optional clause.
```suggestion
 * the network stack, when passing any invalid parameters.
```